### DISCUSSION TITLE
How can i make own vpn with this client

### DISCUSSION THREAD

**Discussion by Emranul-Islam** [schwabe/ics-openvpn#1638] [General]
As I am not good at android development, I'm going to make an VPN app with my own design I just want to use Open vpnVPNclient to my app, How can i include this project as a module on my own project to connect open VPN ???

**Comment by schwabe** [comment]
This project is not designed to build your own VPN. So you will make effort to do so and that will not be an easy task for someone who is not good at Android development. Also keep in mind that this project is GPL licensed, so any app you build upon my project is also GPL license and must be open sourced if published.

**Reply by Emranul-Islam** [reply]
can i convert this into a module ?
yeah I understand if I make any VPN app I will make it open source.

**Comment by qq5050595531** [comment]
I just want to make a VPN app，other guys have another project or module which I can import to my own project recomand ? thank you.
